src/opencl/image_function_opencl.cpp

```diff
@@ -1144,6 +1164,62 @@ namespace Image_Function_OpenCL
         multiCL::launchKernel2D( kernel, width, height );
     }
 
+    void SetPixel( Image & image, uint32_t x, uint32_t y, uint8_t value )
+    {
+        Image_Function::ParameterValidation( image );
+
+        const multiCL::OpenCLProgram & program = GetProgram();
+        multiCL::OpenCLKernel kernel( program, "setPixelOpenCL" );
+
+        if ( image.empty() || x >= image.width() || y >= image.height() )
+            throw imageException( "Bad input parameters in image function" );
+
+        const uint32_t width = image.width();
+        const uint32_t height = image.height();
+        const uint32_t rowSize = image.rowSize();
+        const uint32_t offset = x * rowSize + y;
+
+        kernel.setArgument( image.data(), offset, rowSize, width, height, value );
+
+        multiCL::launchKernel2D( kernel, image.width(), image.height() );
+    }
+
+    void SetPixel( Image & image, const std::vector<uint32_t> & X, const std::vector<uint32_t> & Y, uint8_t value )
+    {
+        Image_Function::ParameterValidation( image );
+
+        const multiCL::OpenCLProgram & program = GetProgram();
+        multiCL::OpenCLKernel kernel( program, "setPixelOpenCL");
+
+        if ( image.empty() || X.empty() || X.size() != Y.size() )
+            throw imageException( "Bad input parameters in image function" );
+
+        const uint32_t width = image.width();
+        const uint32_t height = image.height();
+
+        for ( size_t i = 0; i < X.size(); ++i)
+        {
+            if ( X[i] >= width || Y[i] >= height)
+                throw imageException( "Bad input parameters in image function" );
+        }
+
+        const uint32_t rowSize = image.rowSize();
+        std::vector <uint32_t > O {};
+        for ( size_t i = 0; i < X.size(); ++i)
+        {
+            O.emplace_back(X[i] * rowSize + Y[i]);
+        }
```

Comment:
Remove this code. Offset is only for images.

Reply:
Noted and changed, this is where I was confused a lot. :) can you explain when and where should I use the offset?